src/main/java/org/owasp/esapi/logging/appender/ServerInfoSupplier.java

```diff
@@ -57,10 +58,19 @@ public String get() {
                 appInfo.append(request.getLocalAddr()).append(":").append(request.getLocalPort());
             }
         }
-        if (logAppName) {
-            appInfo.append("/").append(applicationName);
+
+        if (this.logAppName) {
+            if (this.applicationName != null && !this.applicationName.isEmpty()) {
+                appInfo.append("/").append(this.applicationName);
+            }
+            else if (this.applicationName == null) {
+                appInfo.append("/").append(this.applicationName);
+            }
```

Comment:
this feels incorrect.
If the applicationname is null then we want to log it?

Reply:
Looking closer at this,  I'm not sure the inner if/elsif block adds value?

//If not null and not empty -> Log the app name
// else if null -> Log the app name

I think it makes sense to remove the  applicationName validation and  log the applicationName if it's configured to log.

```
if (this.logAppName) {
  appInfo.append("/").append(this.applicationName);
}
```

If the log does not appear correctly, that is a different configuration that should be made in the environment.

Reply:
If `applicationName` is `null` or the empty string, perhaps we should log it as "Unknown"?

Reply:
If that were the case, then we'd want to change the default value returned from the SecurityConfiguration so that an unset ApplicationName would be consistent throughout the instance.
I think that's out of scope for this work, and I think we should log the value that was pulled from the config (even if that is null). 
Note that the ApplicationName is pulled from the SecurityConfiguration in each of the LogFactory implementations:
` String appName = ESAPI.securityConfiguration().getStringProp(APPLICATION_NAME);`

Reply:
@jeremiahjstacey - Fair enough. But maybe something we should keep in mind for the future.